src/app/Plans/components/PlanStatus.tsx

```diff
@@ -1,43 +1,52 @@
 import * as React from 'react';
-import { IMigration, IPlan } from '@app/queries/types';
+import { Progress, ProgressVariant, ProgressMeasureLocation } from '@patternfly/react-core';
 import { StatusIcon, StatusType } from '@konveyor/lib-ui';
-import { Level, LevelItem, Progress, ProgressMeasureLocation, Text } from '@patternfly/react-core';
-import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
 
 interface IPlanStatusProps {
-  plan: IPlan;
-  migration: IMigration;
+  status: string;
+  value?: number;
+  message?: string;
 }
 
 const PlanStatus: React.FunctionComponent<IPlanStatusProps> = ({
-  plan,
-  migration,
+  status,
+  value,
+  message,
 }: IPlanStatusProps) => {
-  if (plan.status.conditions.every((condition) => condition.type === 'Ready')) {
-    return <StatusIcon status={StatusType.Ok} label="Ready" />;
-  } else {
-    // TODO: This is only a placeholder, more statuses check needs to be done.
-    const totalVMs = plan.spec.vmList.length;
-    const percentVMsDone = totalVMs > 0 ? (migration.status.nbVMsDone * 100) / totalVMs : 0;
-    const label = 'Running';
+  let isReady = false;
+  let title = '';
+  let variant: ProgressVariant | undefined;
 
-    return (
-      <>
-        <Level className={`${spacing.mbXs} ${spacing.prMd}`}>
-          <LevelItem>
-            <StatusIcon
-              status={StatusType.Warning}
-              label={<Text component="small">{label}</Text>}
-            />
-          </LevelItem>
-          <LevelItem>
-            <Text component="small">{`${migration.status.nbVMsDone} of ${plan.spec.vmList.length} VMs migrated`}</Text>
-          </LevelItem>
-        </Level>
-        <Progress value={percentVMsDone} measureLocation={ProgressMeasureLocation.none} />
-      </>
-    );
+  switch (status) {
+    case 'Ready':
+      title = 'Ready';
+      isReady = true;
+      break;
+    case 'Error':
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    case 'Finished':
+      title = 'Complete';
+      variant = ProgressVariant.success;
+      break;
+    default:
+      title = 'Running';
+      break;
```

Comment:
I usually avoid `switch` statements because of some issues I ran into in the past... but, I can't really remember why... so maybe this is fine? @ibolton336 do you have any opinions about `switch` vs `if-else` for something like this?

Also I think you don't need a `default` case here, instead you could just have `let title = 'Running'` be the default value above. It seems in that same case you never set a value for `variant`, is that intentional?

Reply:
@mturley, I wasn't happy enough with that piece, I've re-factored .

Reply:
Let's just keep it, I'm realizing my take on it is personal preference. It is weird that it's the only JS syntax that doesn't use brackets around things that seem like blocks :)

Reply:
@mturley, you're right, there was actually some duplicated logic between the condition check between the caller and PlanStatus, I removed the switch.

Reply:
oh, ok cool!

Reply:
@mturley, sorry I added one more to use more types instead of string.